test/unit/sessions/test_attachment_models.py

```diff
@@ -0,0 +1,518 @@
+            manifest_properties=manifest_props, local_root_path="/valid/root"
+        )
+        assert worker_props.local_root_path == "/valid/root"
+
+        # Test that validation still occurs in class method
+        with pytest.raises(ValueError, match="local_root_path must be set"):
+            WorkerManifestProperties.from_manifest_properties(
+                manifest_properties=manifest_props, local_root_path=""
+            )
+
+    def test_local_manifest_paths_default_factory(self):
+        """Test that each instance gets its own local_manifest_paths list (mutable default fix)."""
+        manifest_props1 = ManifestProperties(
+            rootPath="/source/path1", rootPathFormat=PathFormat.POSIX
+        )
+        manifest_props2 = ManifestProperties(
+            rootPath="/source/path2", rootPathFormat=PathFormat.POSIX
+        )
+
+        # Create two instances without specifying local_manifest_paths
+        worker_props1 = WorkerManifestProperties(
+            manifest_properties=manifest_props1, local_root_path="/local/root1"
+        )
+        worker_props2 = WorkerManifestProperties(
+            manifest_properties=manifest_props2, local_root_path="/local/root2"
+        )
+
+        # Verify each instance has its own empty list
+        assert worker_props1.local_manifest_paths == []
+        assert worker_props2.local_manifest_paths == []
+        assert worker_props1.local_manifest_paths is not worker_props2.local_manifest_paths
+
+        # Modify one instance's list and verify the other is unaffected
+        worker_props1.local_manifest_paths.append("/path/to/manifest1.json")
+        assert worker_props1.local_manifest_paths == ["/path/to/manifest1.json"]
+        assert worker_props2.local_manifest_paths == []
+
+        # Modify the second instance's list
+        worker_props2.local_manifest_paths.extend(
+            ["/path/to/manifest2.json", "/path/to/manifest3.json"]
+        )
+        assert worker_props1.local_manifest_paths == ["/path/to/manifest1.json"]
+        assert worker_props2.local_manifest_paths == [
+            "/path/to/manifest2.json",
+            "/path/to/manifest3.json",
+        ]
+
+    def test_local_manifest_paths_initialization_with_provided_list(self):
+        """Test initialization with explicitly provided local_manifest_paths."""
+        manifest_props = ManifestProperties(
+            rootPath="/source/path", rootPathFormat=PathFormat.POSIX
+        )
+
+        provided_paths = ["/provided/manifest1.json", "/provided/manifest2.json"]
+        worker_props = WorkerManifestProperties(
+            manifest_properties=manifest_props,
+            local_root_path="/local/root",
+            local_manifest_paths=provided_paths,
+        )
+
+        # Verify the provided list is used
+        assert worker_props.local_manifest_paths == provided_paths
+        # Verify it's the same reference (not a copy)
+        assert worker_props.local_manifest_paths is provided_paths
```

Comment:
Is this the behaviour we want?

This means that both `WorkerManifestProperties` and the caller can modify the list and affect one another. Usually copy semantics are preferred so that `WorkerManifestProperties` maintains its own separately controlled list and is unaffected by changes made by the caller to the list supplied when creating the instance.